lib/topology/platforms/shell.py

```diff
@@ -285,14 +296,15 @@ def __init__(
         self._timeout = timeout or -1
         self._encoding = encoding
         self._try_filter_echo = try_filter_echo
+        self._auto_connect = auto_connect
```

Comment:
Also, a value of ``None`` in the ``connection`` attribute only means "use the default connection" as defined above:

    #. Every time any operation with the shell is attempted, a connection needs to be specified, if not, the *default* connection of the shell is used.